core/src/main/scala/quasar/plugin/postgres/datasource/PostgresDatasource.scala

```diff
@@ -158,6 +164,22 @@ final class PostgresDatasource[F[_]: Bracket[?[_], Throwable]: MonadResourceErr:
   private def hygienicIdent(ident: Ident): Ident =
     s""""${ident.replace("\"", "\"\"")}""""
 
+  private def maskedColumns: ScalarStages => Option[(NonEmptyList[String], ScalarStages)] = {
+    case ss @ ScalarStages(idStatus, ScalarStage.Mask(m) :: t) =>
+      for {
+        paths <- m.keySet.toList.toNel
+
+        fields0 <- paths.traverse(_.head collect { case CPathField(f) => f })
```

Comment:
Does this just mask the top-level path component? Meaning, if we were masking `.foo.bar.baz` we'd only mask `.foo` and then retain everything underneath it?

Reply:
Yes, exactly.

Reply:
Could you add a test that exemplifies this as well?